### PR TITLE
Add WouldWait to sharedstore getters

### PR DESCRIPTION
**Pull request by lavoiesl**
This can be used by async requests whether to reply right away with "yes I have your data" or "here's a job id, come back later"

### CODE REVIEW

File: sharedstore/getter.go [hunk 1]
```diff
@@ -30,6 +35,10 @@ type promiseGetter struct {
 	promise lockmap.Promise
 }
 
+func (g *promiseGetter) WouldWait() bool {
+	return true
```
Comment (lugray) [inline sharedstore/getter.go]: Have I misunderstood something, or should this be
```suggestion
	select {
	case <-g.promise:
		return false
	case <-ctx.Done():
		return false
	default:
		return true
	}
```

File: sharedstore/getter.go [hunk 2]
```diff
@@ -48,6 +57,10 @@ type pollGetter struct {
 	store Store
 }
 
+func (g *pollGetter) WouldWait() bool {
+	return true
```
Comment (lugray) [inline sharedstore/getter.go]: And should this be similarly smarter?

File: sharedstore/getter.go (hunk 1 above)
Comment (lavoiesl) [reply sharedstore/getter.go]: That is superior indeed, thanks :)

File: sharedstore/getter.go (hunk 2 above)
Comment (lavoiesl) [reply sharedstore/getter.go]: Added a check for the context, but I ultimately don't care about this one, the shared store should not give you a pollGetter if it's not supposed to wait anyway.

**Pull request merged**